client/flutter/lib/pages/symptom_checker/symptom_checker_view.dart

```diff
@@ -0,0 +1,157 @@
+    );
+  }
+
+  Widget _buildPage(BuildContext context) {
+    if (_model == null) {
+      return _buildMessage("Loading...", loading: true);
+    }
+    if (_model.isComplete) {
+      return _buildComplete();
+    }
+    if (_model.seekMedicalAttention) {
+      return _buildMessage("Seek Medical Attention!");
+    }
+    return PageView(
+        physics: NeverScrollableScrollPhysics(),
+        controller: _controller,
+        children: _pages ?? []);
+  }
+
+  Widget _viewForPageModel(SymptomCheckerPageModel model) {
+    switch (model.question.type) {
+      case SymptomCheckerQuestionType.YesNo:
+        return YesNoQuestionView(pageDelegate: this, pageModel: model);
+      case SymptomCheckerQuestionType.ShortListSingleSelection:
+        return ShortListQuestionView(pageDelegate: this, pageModel: model);
+      case SymptomCheckerQuestionType.ShortListMultipleSelection:
+        return ShortListQuestionView(pageDelegate: this, pageModel: model);
+      case SymptomCheckerQuestionType.LongListSingleSelection:
+        return LongListQuestionView(pageDelegate: this, pageModel: model);
+    }
+    throw Exception("can't reach here");
+  }
+
+  Widget _buildComplete() {
+    return Column(
+      children: <Widget>[
+        Spacer(),
+        Center(child: SymptomCheckerSummary(model: _model)),
+        SizedBox(height: 24),
+        FlatButton(
+            color: Colors.grey,
+            child: Text("Dismiss"),
+            onPressed: () {
+              Navigator.pop(context);
+            }),
+        Spacer(flex: 3),
+      ],
+    );
+  }
+
+  Widget _buildMessage(String text, {bool loading = false}) {
+    return Column(
+      children: <Widget>[
+        Spacer(),
+        Center(
+            child: Column(
+          children: <Widget>[
+            Text(text),
+            if (loading)
+              Padding(
+                padding: const EdgeInsets.only(top: 16.0),
+                child: CupertinoActivityIndicator(),
+              ),
+          ],
+        )),
+        Spacer(flex: 3),
+      ],
+    );
+  }
+
+  void _modelChanged() {
+    setState(() {
+      _pages = _model.pages.map(_viewForPageModel).toList();
+    });
+    _nextPage();
+  }
+
+  void _nextPage() {
+    if (!_controller.hasClients) {
+      return;
+    }
+    if (_controller.page < _model.pages.length) {
+      _controller.animateToPage(_model.pages.length - 1,
+          duration: Duration(milliseconds: 500), curve: Curves.easeInOut);
+    }
+  }
+
+  Future<void> _previousPage() => _controller.previousPage(
+      duration: Duration(milliseconds: 500), curve: Curves.easeInOut);
+
+  /// Receive answers from the page and update the model.
+  @override
+  void answerQuestion(Set<String> answerIds) {
+    _model.answerQuestion(answerIds);
+  }
+
+  /// Receive back indication from the page and update the model.
+  @override
+  void goBack() async {
+    await _previousPage();
+    _model.previousQuestion();
```

Comment:
Since the model's `previousQuestion` will notify this class as a listener to call `_nextPage()`, could we simplify this flow and remove the need for the `_previousPage()` call? It feels more unidirectional if the PageController is only ever an expression of the model's last page index. (Not sure if I'm expressing myself clearly here, let me know if this makes sense)

Reply:
This is a good question.  So, the reason I kept the view in the loop here was two-fold:  1) So that the UI can perform arbitrary animated transitions to move to the next/previous page.  2) I wanted to preserve the option for us to allow the user to simply drag the page back to go backwards in the page view rather than having to use the button.  In fact if you simply remove the `NeverScrollableScrollPhysics` it will allow you to do this.  I just did not hook up logic to notice that and inform the model to revert a page.

TLDR: I was really unclear on what kind of navigation the designers are going to want for this and I wanted to keep it flexible.  

A related topic - If you go back and then immediately go forward again should it preserve your previous choice on the second page or clear it?  Right now it is clearing it, but technically it doesn't have to unless choice of question changes.

Reply:
Makes a lot of sense - to the first point, could we handle arbitrary transitions in the `_nextPage` call? But, I think it makes sense to keep it the way you've implemented for now, and we can always re-examine when we know more about the transitions.

@davidkaneda & @asianfoodlab - see Pat's question above about whether going back and then forward should preserve answer choices, would love to get your thoughts! One thing to keep in mind is that in the future, when/if we have conditional logic on whether or not to show questions, we may not even show the same question if the user goes back and changes their answer.

Reply:
Would bank on saving the current answers if the user goes back

Reply:
@hspinks re: `_nextPage` So, the thing that is currently muddying this is that the model has any notion of the "current page"...  We can change the API to let the view update any page of the model with new answers and have it simply do the right thing (e.g. change or truncate subsequent answers).  However the view will still need to participate in the user's responses in order to maintain its own notion of the current page.  e.g. take the case where the user goes back a page and then goes forward again without making changes... In that case there is no change to the model, only the view's notion of the current page.  So the view will always need to be in the loop and this makes sense if you think of it as a "controller" (in the classic MVC sense).

Reply:
Makes sense to me. I think that if we do want to update this to make sure the model is the source of truth of the current page, then that could happen in a different PR when/if we feel like we have a need for that